src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,51 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ELEMENTS_NUMBER = 10;
+    private final K[] keys;
+    private final V[] values;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[MAX_ELEMENTS_NUMBER];
+        this.values = (V[]) new Object[MAX_ELEMENTS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
+            this.values[getKeyIndex(key)] = value;
+            return;
+        }
+        int lastIndex = size();
+        this.keys[lastIndex] = key;
+        this.values[lastIndex] = value;
     }
 
     @Override
     public V get(K key) {
-        return null;
+        return getKeyIndex(key) >= 0
+                ? values[getKeyIndex(key)]
+                : null;
```

Comment:
```suggestion
        return getKeyIndex(key) >= 0 ? values[getKeyIndex(key)] : null;
```
If you can call another method just once it is better to do so